tools/shared.py

```diff
@@ -2427,8 +2429,8 @@ def calculate_reachable_functions(infile, initial_list, can_reach=True):
   def closure_compiler(filename, pretty=True, advanced=True, extra_closure_args=[]):
     with ToolchainProfiler.profile_block('closure_compiler'):
       if not check_closure_compiler():
-        logger.error('Cannot run closure compiler')
-        raise Exception('closure compiler check failed')
+        logger.error('google-closure-compiler executable was not found: Cannot run closure compiler')
```

Comment:
Replace these to lines with `exit_with_error`?

Also, I think we should perhaps add the path where it is expected to be, and perhaps something like `if you not using emsdk, you may need to run "npm install"`?

Reply:
Check for `exit_with_error`. The other bit of information is already printed inside `check_closure_compiler`, so no need repeating here.